birdhouse/components/stac/config/proxy/conf.extra-service.d/stac.conf.template

```diff
@@ -9,7 +9,9 @@
         proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
         proxy_set_header X-Forwarded-Host $host:$server_port;
         proxy_buffering off;
+        set $access_control_allow_origin $stac_origin_allowed;
         include /etc/nginx/conf.d/cors.include;
+        add_header Vary Origin;
```

Comment:
What does the Vary header do again?  Why we need to add it here?

Reply:
The thing though is that `Vary` should be added only if the `Origin` (or any other header) actually affects the response (assuming no other header already distinguishes the response like the `Content-Type`). Adding it systematically (eg: when `Access-Control-Allow-Origin: *`) could result in duplicate cache entries for each domain accessing the server although nothing changes between them. It is used purposely to generate a "diff" in the cached response according to its domain-specific value, so that each domain is managed separately. I think we don't have a choice to apply it on case-by-case basis.

Reply:
> Adding it systematically (eg: when `Access-Control-Allow-Origin: *`)

I was suggesting to add `Vary` only when `Access-Control-Allow-Origin` is **not** `*`.  Hopefully the if-statement in Nginx allows this.  Otherwise we should put a big reminder somewhere so other users notice it.

Reply:
> `add_header Vary "";` the empty value won't cause any side-effect and is a legit value allowed by HTTP?

I just re-read Francis response earlier (https://github.com/bird-house/birdhouse-deploy/pull/599#discussion_r2482567612) ? "nginx omits header if empty".  Could you add this comment to the change so it is clear for other readers too.

Reply:
For sure! Added in https://github.com/bird-house/birdhouse-deploy/pull/599/commits/2aa0caed5087250ed7682c9cc5c3b9cc46573dc7